core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -85,6 +85,10 @@ private[spark] class SparkSubmit extends Logging {
         Logging.disableStructuredLogging()
       }
     }
+
+    // We should initialize log again after `spark.log.structuredLogging.enabled` effected
+    Logging.uninitialize()
```

Comment:
@pan3793 sorry for the late reply. We should fix this issue for sure. However, explicitly calling `Logging.uninitialize()` here seems weird. 
I wonder if we can have a better solution than this.

Reply:
`Logging.uninitialize()` is designed and was used to re-initialize the logging system

https://github.com/apache/spark/blob/168230a812870b51d84e0a2bd3d10cbba4994eaf/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L960-L963

Reply:
@gengliangwang do you have better suggestions?

Reply:
I will get back to this one in one day.

Reply:
ok, I will merge this one once the unused variable is removed